rust/connlib/tunnel/src/io/udp_dns.rs

```diff
@@ -0,0 +1,40 @@
+use std::{
+    io,
+    net::{Ipv4Addr, Ipv6Addr, SocketAddr},
+    sync::Arc,
+};
+
+use domain::base::{Message, ToName as _};
+use socket_factory::{SocketFactory, UdpSocket};
+
+pub async fn send(
+    factory: Arc<dyn SocketFactory<UdpSocket>>,
+    server: SocketAddr,
+    query: Message<Vec<u8>>,
+) -> io::Result<Message<Vec<u8>>> {
+    let domain = query
+        .sole_question()
+        .expect("all queries should be for a single name")
+        .qname()
+        .to_vec();
+    let bind_addr = match server {
+        SocketAddr::V4(_) => SocketAddr::new(Ipv4Addr::UNSPECIFIED.into(), 0),
+        SocketAddr::V6(_) => SocketAddr::new(Ipv6Addr::UNSPECIFIED.into(), 0),
+    };
+
+    tracing::trace!(target: "wire::dns::recursive::udp", %server, %domain);
+
+    // To avoid fragmentation, IP and thus also UDP packets can only reliably sent with an MTU of <= 1500 on the public Internet.
+    const BUF_SIZE: usize = 1500;
```

Comment:
In many cases this will be lower though, for example on GCP it's 1460 IIRC. Just in case it matters

Reply:
Doesn't matter, this is just the upper bound :)